notification/twilio/sms.go

```diff
@@ -244,6 +244,13 @@ func (s *SMS) ServeMessage(w http.ResponseWriter, req *http.Request) {
 		}
 
 		if isPassive {
+			valid, err := s.r.IsKnownDest(ctx, from)
+			if err != nil {
+				log.Log(ctx, fmt.Errorf("check if known SMS number: %w", err))
+			} else if !valid {
+				// don't respond if the number is not known
+				return
+			}
```

Comment:
Yes, but we can discuss if that should be the case. In the case of a DB/network issue, it will fall back to sending the reply and only blocks it on a sucessful negative check.